controllers/keda/hpa.go

```diff
@@ -293,7 +287,7 @@ func (r *ScaledObjectReconciler) getScaledObjectMetricSpecs(ctx context.Context,
 					Metric: autoscalingv2.MetricIdentifier{
 						Name: compMetricName,
 						Selector: &metav1.LabelSelector{
-							MatchLabels: map[string]string{kedav1alpha1.ScaledObjectOwnerAnnotation: scaledObject.Name},
+							MatchLabels: map[string]string{kedav1alpha1.ScaledObjectOwnerAnnotation: scaledObjectName},
```

Comment:
I'm not 100% sure if we should do this or just print an error. We use [this label in to recover the SO name during the HPA querying](https://github.com/kedacore/keda/blob/main/pkg/provider/provider.go#L99) and I'd say that the querying will fail if we truncate this name. But maybe I'm wrong and if we cover the scenario with and e2e test to cover it, I'm fine 😄

Reply:
oddly enough, HPA seems to have the label [truncated since 2021](https://github.com/kedacore/keda/pull/1631), there it's probably not used that much.

For sure we should add e2e tests for trunc name cases, I'd be curious to see what is the behaviour when two SO with names 63+ chars that differ only in the last truncated characters.

Alternatively, can we refactor the code to use SO `uid` field? that is guaranteed to be within the bounds but not sure if that would be a breaking change.

Reply:
> oddly enough, HPA seems to have the label https://github.com/kedacore/keda/pull/1631, there it's probably not used that much.

That applies to the HPA name, and this label is this other line -> https://github.com/kedacore/keda/pull/1631/files#diff-a43846d8ca384145be0f11e006bc0ce798b50c992df46e9b13ff0c55c290bae0L157

Reply:
The HPA name isn't relevant if there isn't more HPAs in conflict in the namespace, but the ScaledObject name is used to get the SO

Reply:
So, do we move forward by simply printing an error and relying on an end-to-end job to cover this?

Reply:
as @JorTurFer mentioned, the `ScaledObject` will be patched with a truncated label now but getting metrics will probably not work because of how it's implemented in the [`provider.go`](https://github.com/kedacore/keda/blob/c43af597fc94c8ffa8f444557d2647fc2eb1c318/pkg/provider/provider.go#L99). I don't have any good ideas how to work around that unless we refactor the internals to not use the `name` but `uid` for example.

Reply:
I like @wozniakjan's suggestion